src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1372,71 +1365,65 @@ public bool ContainsFile(string fullPath)
 
             public void ProcessFileChange(string filePath)
             {
-                ProcessFileChange(filePath, filePath);
+                ProcessFileChange(filePath, dynamicFileInfoOpt: null);
             }
 
             /// <summary>
             /// Process file content changes
             /// </summary>
-            /// <param name="projectSystemFilePath">filepath given from project system</param>
             /// <param name="workspaceFilePath">filepath used in workspace. it might be different than projectSystemFilePath. ex) dynamic file</param>
-            public void ProcessFileChange(string projectSystemFilePath, string workspaceFilePath)
+            /// <param name="dynamicFileInfoOpt">dynamic file that has changed</param>
+            public void ProcessFileChange(string workspaceFilePath, DynamicFileInfo dynamicFileInfoOpt)
```

Comment:
Warning: this will collide with #31574. You may want to revise this after this goes in since the that resulted in some more general flow changes to how this works.